** Not yet complete **
RCS file: /cvs/external/python/makefile.mk,v
retrieving revision 1.3.2.2
Index: config_office/configure.in
===================================================================
--- config_office/configure.in.orig	2004-04-04 16:20:55.000000000 +0200
+++ config_office/configure.in	2004-04-04 16:22:34.000000000 +0200
@@ -1470,12 +1470,38 @@ if test -n "$with_python"; then
         PYTHON="$with_python"
     fi
     AM_PATH_PYTHON([2.2])
+
+    py_prefix=`$PYTHON -c "import sys; print sys.prefix"`
+    py_exec_prefix=`$PYTHON -c "import sys; print sys.exec_prefix"`
+
+    PYTHONHOME="$py_prefix"
+      PYTHON_CFLAGS="-I$py_prefix/include/python$PYTHON_VERSION"
+    if test "$py_prefix" != "$py_exec_prefix"; then
+      PYTHONHOME="$PYTHONHOME:$py_exec_prefix"
+      PYTHON_CFLAGS="$PYTHON_CFLAGS -I$py_exec_prefix/include/python$PYTHON_VERSION"
+    fi
+    dnl check if the headers exist:
+    save_CPPFLAGS="$CPPFLAGS"
+    CPPFLAGS="$CPPFLAGS $PYTHON_CFLAGS"
+    AC_CHECK_HEADER(Python.h, [],
+        [AC_MSG_ERROR(Python.h not found. Install python headers/development package.)], [])
+    CPPFLAGS="$save_CPPFLAGS"
+
+    PYMAJOR=$($PYTHON -c "import sys; print sys.version_info[[0]]")
+    PYMINOR=$($PYTHON -c "import sys; print sys.version_info[[1]]")
+    PYMICRO=$($PYTHON -c "import sys; print sys.version_info[[2]]")
+    PYVERSION=$($PYTHON -c "import sys; print '%d.%d.%d' % sys.version_info[[:3]]")
 else
     AC_MSG_RESULT([internal])
     SCPDEFS="$SCPDEFS -DWITH_PYTHON"
 fi
 AC_SUBST(PYTHON)
-AC_SUBST(PYTHON_VERSION)
+AC_SUBST(PYTHONHOME)
+AC_SUBST(PYTHON_CFLAGS)
+AC_SUBST(PYMAJOR)
+AC_SUBST(PYMINOR)
+AC_SUBST(PYMICRO)
+AC_SUBST(PYVERSION)
 
 dnl ===================================================================
 dnl Check for system zlib
Index: config_office/set_soenv.in
===================================================================
--- config_office/set_soenv.in.orig	2004-04-04 16:20:55.000000000 +0200
+++ config_office/set_soenv.in	2004-04-04 16:22:34.000000000 +0200
@@ -1352,7 +1352,14 @@ else
 }
 
 $PYTHONPATH           = '.'.$ps.'$SOLARVER'.$ds.'$UPD'.$ds.'$INPATH'.$ds.'lib'.$ps.'$SOLARVER'.$ds.'$UPD'.$ds.'$INPATH'.$ds.'lib'.$ds.'python'.$ps.'$SOLARVER'.$ds.'$UPD'.$ds.'$INPATH'.$ds.'lib'.$ds.'python'.$ds.'lib-dynload';
-$PYTHONHOME           = '$SOLARVER'.$ds.'$UPD'.$ds.'$INPATH';
+if ("@PYTHON@" eq "")
+{
+  $PYTHONHOME         = '$SOLARVER'.$ds.'$UPD'.$ds.'$INPATH';
+}
+else
+{
+  $PYTHONHOME         = '@PYTHONHOME@';
+}
 
 #
 print "done\n";
@@ -1779,7 +1786,11 @@ ToFile( "LIBSN_LIBS",        "@LIBSN_LIB
 ToFile( "WITH_MOZILLA",      "@WITH_MOZILLA@",     "e" );
 ToFile( "WITH_FONTS",        "@WITH_FONTS@",       "e" );
 ToFile( "SYSTEM_PYTHON",     "@PYTHON@",           "e" );
-ToFile( "PYTHON_VERSION",    "@PYTHON_VERSION@",   "e" );
+ToFile( "PYTHON_CFLAGS",     "@PYTHON_CFLAGS@",    "e" );
+ToFile( "PYMAJOR",           "@PYMAJOR@",          "e" );
+ToFile( "PYMINOR",           "@PYMINOR@",          "e" );
+ToFile( "PYMICRO",           "@PYMICRO@",          "e" );
+ToFile( "PYVERSION",         "@PYVERSION@",        "e" );
 if ( '@ENABLE_RPATH@' eq "no" ) {
   ToFile( "LINKFLAGSRUNPATH",'',                   "e" );
 }
Index: python/makefile.mk
===================================================================
--- python/makefile.mk.orig	2004-04-04 16:20:55.000000000 +0200
+++ python/makefile.mk	2004-04-04 16:22:34.000000000 +0200
@@ -155,6 +155,9 @@ PYVERSIONFILE=$(MISC)$/pyversion.mk
 PYCONFIG=$(MISC)$/build$/pyconfig.h
 .ENDIF
 
+# Only build this project if there is no SYSTEM PYTHON
+.IF "$(SYSTEM_PYTHON)" == ""
+
 .INCLUDE : set_ext.mk
 .INCLUDE : target.mk
 .INCLUDE : tg_ext.mk
@@ -231,3 +234,7 @@ $(PYVERSIONFILE) : pyversion.mk $(PACKAG
 
 patch : $(MISC)$/convert_dos_flag
 
+.ELSE
+# Don't build
+dummy :
+.ENDIF
Index: pyuno/source/loader/makefile.mk
===================================================================
--- pyuno/source/loader/makefile.mk.orig	2004-04-04 16:20:55.000000000 +0200
+++ pyuno/source/loader/makefile.mk	2004-04-04 16:22:34.000000000 +0200
@@ -69,10 +69,16 @@ ENABLE_EXCEPTIONS=TRUE
 
 DLLPRE = 
 
+.IF "$(SYSTEM_PYTHON)" == ""
 .INCLUDE :  pyversion.mk
+.ENDIF
 #-------------------------------------------------------------------
 
+.IF "$(SYSTEM_PYTHON)" == ""
 CFLAGS+=-I$(SOLARINCDIR)$/python
+.ELSE
+CFLAGS+="$(PYTHON_CFLAGS)"
+.ENDIF
 
 .IF "$(OS)$(CPU)$(COMEX)" == "SOLARISS4"
 # no -Bdirect for SunWS CC
@@ -81,8 +87,12 @@ DIRECT = $(LINKFLAGSDEFS)
 
 .IF "$(GUI)" == "UNX"
 PYUNOLIB=-lpyuno
+.IF "$(SYSTEM_PYTHON)" == ""
 PYTHONLIB=-lpython
 .ELSE
+PYTHONLIB=-lpython$(PYMAJOR).$(PYMINOR)
+.ENDIF
+.ELSE
 PYUNOLIB=ipyuno.lib
 PYTHONLIB=python$(PYMAJOR)$(PYMINOR).lib
 .ENDIF
Index: pyuno/source/module/makefile.mk
===================================================================
--- pyuno/source/module/makefile.mk.orig	2004-04-04 16:20:55.000000000 +0200
+++ pyuno/source/module/makefile.mk	2004-04-04 16:22:34.000000000 +0200
@@ -66,10 +66,17 @@ ENABLE_EXCEPTIONS=TRUE
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :  settings.mk
+.IF "$(SYSTEM_PYTHON)" == ""
 .INCLUDE :  pyversion.mk
+.ENDIF
 #-------------------------------------------------------------------
 
+.IF "$(SYSTEM_PYTHON)" == ""
 CFLAGS+=-I$(SOLARINCDIR)$/python
+.ELSE
+CFLAGS+="$(PYTHON_CFLAGS)"
+.ENDIF
+
 .IF "$(OS)$(CPU)$(COMEX)" == "SOLARISS4"
 # no -Bdirect for SunWS CC
 DIRECT = $(LINKFLAGSDEFS)
Index: pyuno/zipcore/makefile.mk
===================================================================
--- pyuno/zipcore/makefile.mk.orig	2004-04-04 16:20:55.000000000 +0200
+++ pyuno/zipcore/makefile.mk	2004-04-04 16:22:34.000000000 +0200
@@ -2,6 +2,8 @@ PRJNAME=pyuno
 PRJ=..
 
 .INCLUDE : settings.mk
+
+.IF $(SYSTEM_PYTHON) == ""
 .INCLUDE : pyversion.mk
 
 PYDIRNAME=python-core-$(PYVERSION)
@@ -16,10 +18,17 @@ FILES=\
 	$(PYTHONBINARY)	\
 	$(foreach,i,$(FINDLIBFILES) $(DESTROOT)$/lib$(i)) 
 
-target: \
+	target: \
 	$(BIN)$/python-core-$(PYVERSION).zip \
 	$(BIN)$/python.sh
 
+.ELSE
+
+target: \
+	$(BIN)$/python.sh
+
+.ENDIF # $(SYSTEM_PYTHON) == ""
+
 $(BIN)$/python.sh : python.sh
 	-rm -f $@
 	cat $? > $@
@@ -27,6 +36,8 @@ $(BIN)$/python.sh : python.sh
 	chmod +x $@
 .ENDIF
 
+.IF $(SYSTEM_PYTHON) == ""
+
 $(BIN)$/python-core-$(PYVERSION).zip : $(FILES)
 .IF "$(GUI)" == "UNX"
 .IF "$(OS)" != "MACOSX"
@@ -51,3 +62,5 @@ $(DESTROOT)$/bin$/python$(EXECPOST) : $(
 .ENDIF
 	chmod +x $@
 .ENDIF
+
+.ENDIF # $(SYSTEM_PYTHON) == ""
Index: scp/source/python/python.scp
===================================================================
--- scp/source/python/python.scp.orig	2004-04-04 16:20:55.000000000 +0200
+++ scp/source/python/python.scp	2004-04-04 16:24:30.000000000 +0200
@@ -188,6 +188,7 @@ ProfileItem GID_PROFILEITEM_PYUNO_UNO_SE
     Value  = "?$PYUNO_USER_PACKAGES/services.rdb ?$PYUNO_SHARED_PACKAGES/services.rdb $ORIGIN/services.rdb";
 End
 
+#ifdef WITH_PYTHON
 File gid_File_Py_python_core
      TXT_FILE_BODY;
      ModuleID = GID_MODULE_PYUNO;
@@ -195,6 +196,7 @@ File gid_File_Py_python_core
      Name = "python-core-2.2.2.zip";
      Styles          = (ARCHIVE);
 End
+#endif
 
 Profile gid_Profile_pythonloader_uno_ini
     ModuleID = GID_MODULE_PYUNO;
@@ -230,6 +232,7 @@ ProfileItem GID_PROFILEITEM_PYTHONLOADER
 End
 
 
+#ifdef WITH_PYTHON
 #ifdef WNT
 File gid_File_Python_dll
      TXT_FILE_BODY;
@@ -282,7 +285,8 @@ Shortcut gid_Shortcut_Lib_Python_2
     Name = STRING(CONCAT2(libpython,UNXSUFFIX));
     Styles = (NETWORK,RELATIVE);
 End
-
+#endif
+#endif // ifdef WITH_PYTHON
 
 File gid_File_Pyuno
     TXT_FILE_BODY;
@@ -300,4 +304,3 @@ File gid_File_python_sh
      Styles = (PACKED);
 End
 
-#endif
Index: scp/source/python/unxbasic_python.scp
===================================================================
--- scp/source/python/unxbasic_python.scp.orig	2004-04-04 16:20:55.000000000 +0200
+++ scp/source/python/unxbasic_python.scp	2004-04-04 16:22:34.000000000 +0200
@@ -70,9 +70,11 @@ Procedure GID_PROCEDURE_PYTHON_INSTALL
     ModuleID = GID_MODULE_PYUNO;
     Code = {
     Sub PythonLink
+#ifdef WITH_PYTHON
         shell("/bin/sh -c 'test -d python-core-2.2.2 && ln -s python-core-2.2.2 python-core'")
-        shell("/bin/sh -c 'test -f python.sh && ln -s python.sh python'")
         shell("/bin/sh -c 'test -f python-core-2.2.2/bin/python && chmod +x python-core-2.2.2/bin/python'")
+#endif
+        shell("/bin/sh -c 'test -f python.sh && ln -s python.sh python'")
 
 	End Sub
 	};
